### ISSUE TITLE
Strange view of single ontology trees

### ISSUE THREAD

**Issue by jmcmurry** [EBISPOT/OLS#102]
The tree views of some single ontologies in OLS (eg, [MP](http://www.ebi.ac.uk/ols/ontologies/mp)) are rendered in a strange way. For comparison see [MP in MGI](http://www.informatics.jax.org/searches/MP_form.shtml). I am struggling to even describe it (eg, lack of labels, subclasses, lots of terms I wouldn't expect to see near the root). Same patterns seen in rendering of [HP](http://www.ebi.ac.uk/ols/ontologies/hp) as well.

MP in OLS | MP in MGI
------|-----
<img width="338" alt="screen shot 2016-07-22 at 12 38 35 pm" src="https://cloud.githubusercontent.com/assets/3658863/17069038/403024ca-5009-11e6-807b-49c345aeeac7.png"> | <img width="507" alt="screen shot 2016-07-22 at 12 36 24 pm" src="https://cloud.githubusercontent.com/assets/3658863/17069042/46578e9c-5009-11e6-9326-4b2b918864be.png">

cc: @cmungall, @mellybelly

You may want to give this issue a more useful title when you figure out what is behind it.

**Comment by cmungall** [comment]
There's no problem with OLS here, I expect the Protege view of mp.owl is similar

import modules require active management and attention. But even if they are working as expected, they can get in the way if you want to look at the main ontology. There is a tension between how much axiomatization to put in the main release and how much to dumb things down. In many cases, the dumb underaxiomatized version goes in the .obo with the axiomatized version in the .owl, but people are increasingly consuming the owl.

We need a combination of best practices, metadata and tuning of the ontology browsers. For example, ontologies within the import closure could be tagged as being core vs support. Difficult issue

**Comment by jmcmurry** [comment]
Ah. Makes sense.

> tags for core vs support

+1; Building on that would make it possible toggle between a "simple" and an "advanced" view in OLS, which would be very welcome for those users outside the ontologencia.

**Comment by cmungall** [comment]
On 22 Jul 2016, at 14:20, Julie McMurry wrote:

> Ah. Makes sense.
>
>> tags for core vs support
>
> +1; Building on that would make it possible toggle between a "simple" 
> and an "advanced" view in OLS, which would be very welcome for those 
> users outside the ontologencia.

Just a warning that this isn't something that can be thrown together. 
What is simple vs advanced is community specific and often depends on 
the tooling. Luckily OLS does the right thing (possibly with a little 
hardcoding) at the moment in most cases. For example, in the 
hierarchical view, is_a and part_of is shown, but not X SubClassOf 
has_part some Y (good: showing this inevitably confused many people). 
This means that we can get relatively complex axioms into the main 
ontology without worrying it will render OLS useless for ordinary 
biologists (you can see the advanced stuff off to the side in infoboxes 
which is perfect). However, imports and modules remain to be tackled.

**Comment by simonjupp** [comment]
One thing we could support is the use of a "root classes" annotations. Ontology owners could either annotate their root classes or provide a list of classes in the config. Then by default the tree view would only show paths to the specific roots (with a toggle to show all). We've had this request from other places too where users don't want to see the top level BFO classification by default. 

Another idea is too have a toggle for only showing terms in the default namespace. This is essentially trying to generate the OBO simple view on the fly. 

We will do some work on this, I suspect we will be able to support both options and simply allow users to choose what kind of view they want. I also want to add support for the collapsed tree view a la AmiGO.

**Comment by cmungall** [comment]
On 27 Jul 2016, at 2:34, simonjupp wrote:

> One thing we could support is the use of a "root classes" annotations. 
> Ontology owners could either annotate their root classes or provide a 
> list of classes in the config. Then by default the tree view would 
> only show paths to the specific roots (with a toggle to show all). 
> We've had this request from other places too where users don't want to 
> see the top level BFO classification by default.

This would work very well

> Another idea is too have a toggle for only showing terms in the 
> default namespace. This is essentially trying to generate the OBO 
> simple view on the fly.

This would presumably be simpler to implement

One concern is that this would further entrench resistance to reuse of 
other ontology classes directly within another ontology. For example, if 
HPO wanted to insert NCIT classes as subclasses of existing classes, 
portions of the hierarchy would disappear when looking at 'HPO'.

The couterargument is that this is already well-entrenched.

> We will do some work on this, I suspect we will be able to support 
> both options and simply allow users to choose what kind of view they 
> want. I also want to add support for the collapsed tree view a la 
> AmiGO.

Great.
